lab_2_tokenize_by_bpe/main.py

```diff
@@ -26,66 +37,153 @@ def collect_frequencies(
     :param end_of_word: a token that signifies the end of word
     :return: dictionary in the form of <preprocessed word: number of occurrences>
     """
+    if not isinstance(text, str) or not isinstance(end_of_word, str) or not (
+            isinstance(start_of_word, str) or start_of_word is None):
+        return None
+    dict_frequencies = {}
+    for i in text.split():
+        word = prepare_word(i, start_of_word, end_of_word)
+        if not word:
+            return None
+        dict_frequencies.update({word: text.split().count(i)})
```

Comment:
you set the same value for the same key many many times.

Reply:
done